application/src/main/java/org/togetherjava/tjbot/commands/moderation/AuditCommand.java

```diff
@@ -139,26 +149,78 @@ private boolean handleChecks(@NotNull Member bot, @NotNull Member author,
         return ModerationUtils.handleHasAuthorRole(ACTION_VERB, hasRequiredRole, author, event);
     }
 
-    private void auditUser(@NotNull User user, @NotNull ISnowflake guild,
-            @NotNull IReplyCallback event) {
-        List<ActionRecord> actions =
-                actionsStore.getActionsByTargetAscending(guild.getIdLong(), user.getIdLong());
+    private @NotNull List<List<ActionRecord>> groupActions(@NotNull List<ActionRecord> actions) {
+        List<List<ActionRecord>> groupedActions = new ArrayList<>();
+        for (int i = 0; i < actions.size(); i++) {
+            if (i % AuditCommand.MAX_PAGE_LENGTH == 0) {
+                groupedActions.add(new ArrayList<>(AuditCommand.MAX_PAGE_LENGTH));
+            }
 
-        MessageEmbed summary = createSummaryMessage(user, actions);
-        if (actions.isEmpty()) {
-            event.replyEmbeds(summary).queue();
-            return;
+            groupedActions.get(groupedActions.size() - 1).add(actions.get(i));
         }
 
-        // Computing messages for actual actions is done deferred and might require asking the
-        // Discord API
-        event.deferReply().queue();
-        JDA jda = event.getJDA();
+        return groupedActions;
+    }
 
-        RestAction<List<MessageEmbed>> messagesTask = RestAction
-            .allOf(actions.stream().map(action -> actionToMessage(action, jda)).toList());
-        messagesTask.map(messages -> prependElement(summary, messages))
-            .flatMap(messages -> event.getHook().sendMessageEmbeds(messages))
-            .queue();
+    private @NotNull Message auditUser(long guildId, long targetId, long callerId, int pageNumber,
+            @NotNull JDA jda) {
+        List<ActionRecord> actions = actionsStore.getActionsByTargetAscending(guildId, targetId);
+        List<List<ActionRecord>> groupedActions = groupActions(actions);
+        int totalPages = groupedActions.size();
+
+        EmbedBuilder audit = createSummaryEmbed(jda.retrieveUserById(targetId).complete(), actions);
+
+        if (groupedActions.isEmpty()) {
+            return new MessageBuilder(audit.build()).build();
+        }
+
+        groupedActions.get(pageNumber - 1)
+            .forEach(action -> audit.addField(actionToField(action, jda)));
+
+        return new MessageBuilder(audit.setFooter("Page: " + pageNumber + "/" + totalPages).build())
+            .setActionRows(makeActionRow(guildId, targetId, callerId, pageNumber, totalPages))
+            .build();
+    }
+
+    private @NotNull ActionRow makeActionRow(long guildId, long targetId, long callerId,
+            int pageNumber, int totalPages) {
+        String stringGuildId = String.valueOf(guildId);
+        String stringTargetId = String.valueOf(targetId);
+        String stringCallerId = String.valueOf(callerId);
+        String stringPageNumber = String.valueOf(pageNumber);
```

Comment:
These extra variables feel obsolete and overly verbose. Due to having a method, you already gave each of the vars a good name. So there is no need to do it again. Inline those please.

Reply:
i did that because i had to use them twice, for both buttons